plugins/inputs/intel_rdt/intel_rdt.go

```diff
@@ -250,8 +255,15 @@ func (r *IntelRDT) createArgsAndStartPQOS(ctx context.Context) {
 func (r *IntelRDT) readData(ctx context.Context, args []string, processesPIDsAssociation map[string]string) {
 	r.wg.Add(1)
 	defer r.wg.Done()
-
-	cmd := exec.Command(r.PqosPath, append(args)...)
+	command := r.PqosPath
+	if r.UseSudo {
+		// Command switched to sh with sudo, note entire command is one string for sudo
+		// must replace ';' with '\;' to allow the shell to pass the command correctly
+		args = []string{"-c", fmt.Sprintf("sudo %s %s", r.PqosPath, strings.Replace(strings.Join(args, " "), ";", "\\;", -1))}
+		command = "/bin/sh"
+		r.Log.Info(fmt.Sprintf("Running pqos with sudo: cmdline=%s %s", command, strings.Join(args, " ")))
```

Comment:
Can we please make this a debug message as it will flood the log otherwise.
```suggestion
		r.Log.Debugf("Running pqos with sudo: cmdline=%s %s", command, strings.Join(args, " "))
```